Added example of icon button w/ dropdown to Navbar component

mrazian85 opened the pull request:
[Link to updated components in Deploy Preview](https://deploy-preview-51--neo-react-library-storybook.netlify.app/?path=/story/components-top-navigation--icon-button-dropdown-example)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Added an example of icon button with dropdown to Top Navigation. No additions to library required; it works using existing Components and updated CSS.

**EXAMPLE FUNCTIONALITY GIF:**
![Navbar icon button dropdown](https://user-images.githubusercontent.com/61242168/196260354-bf3bb0f8-a94e-48f7-8e96-2dc02bb73231.gif)

**IMPORTANT NOTES:**
- Most changes in PR are from dependency updates

yangAtSpoken commented on src/components/TopNav/TopNav.stories.tsx:
```diff
@@ -389,3 +389,36 @@ export const DesignPortalExample = () => (
   </TopNav>
 );
+
+export const IconButtonDropdownExample = () => (
```
Please render and test this in unit test file.

yangAtSpoken reviewed (commented):
one comment

mrazian85 commented:
@yangAtSpoken I have added a test for the new Story; however, it just so happens that this test seems to be triggering an error caused by an upstream issue with `axe-core`.

So I have added the workaround mentioned [here](https://github.com/nickcolley/jest-axe/issues/147) to line 163 of the `TopNav.test.jsx` file.

Please have another look when you get the chance, thanks

matthewAvaya commented:
My comments are below:
- Noticed the icons for the bell and settings has a chevron on hover. Those should be hidden.
- There should be no padding in between each icon button. 

<img width="503" alt="Screen Shot 2022-10-17 at 12 34 16 PM" src="https://user-images.githubusercontent.com/90473510/196266923-bae8d217-ebf2-47e5-96cc-3439b640beea.png">
<img width="523" alt="Screen Shot 2022-10-17 at 12 33 51 PM" src="https://user-images.githubusercontent.com/90473510/196266925-67ee9066-52ec-4718-a0bc-f25f8831b215.png">
<img width="492" alt="Screen Shot 2022-10-17 at 12 33 41 PM" src="https://user-images.githubusercontent.com/90473510/196266929-54b046f7-ecc1-4ee5-8503-f92de7f18d57.png">

yangAtSpoken commented on src/components/TopNav/TopNav.test.jsx:
```diff
@@ -155,5 +156,15 @@ describe("TopNav", () => {
+    describe("Icon Button Dropdown Example", () => {
+      it("passes basic axe compliance", async () => {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        window.getComputedStyle = () => {};
```
looks good.

mrazian85 commented:
@matthewAvaya whoops; good catch on the extra chevrons; I've removed those.

Re: the left padding on the icon button w/ dropdown; I have changed that here but, I will not be able to change it in the CSS library until we publish again, as the PR there has already been approved/merged

matthewAvaya commented:
@mrazian85 Sounds good. Thanks for the info.
